### PR TITLE
Fix: use JID from credentials when credentials_url is set

### PR DESCRIPTION
**Pull request by s-sokolko**
I am not sure if I can make tests for this, but this PR is an attempt to fix https://github.com/conversejs/converse.js/issues/1635

Works for me

### CODE REVIEW

File: src/headless/converse-core.js [hunk 1]
```diff
@@ -1272,6 +1272,9 @@ _converse.initialize = async function (settings, callback) {
             const password = _.isNil(credentials)
                 ? _converse.connection.pass || this.password
                 : credentials.password;
+            const jid = _.isNil(credentials)
+                ? _converse.connection.authcid || this.jid
+                : credentials.jid;
```
Comment (jcbrand) [inline src/headless/converse-core.js]: Why do you add `_converse.connection.authcid` there as well?

Comment (s-sokolko) [reply src/headless/converse-core.js]: @jcbrand I am quite new to Strophe and I tried to make my code similar to password obtaining process (from Strophe connection). I can just use 
```javascript
const jid = _.isNil(credentials)
                ? this.jid
                : credentials.jid;
```
Is it a better option?

Comment (jcbrand) [reply src/headless/converse-core.js]: Yes, I don't think authcid is necessary

Comment (s-sokolko) [reply src/headless/converse-core.js]: @jcbrand done. But tests are failing for some strange reason

**Comment by jcbrand** [comment]
Well done, thanks!

**Pull request merged**